file_formats/cmt_format.mli

```diff
@@ -50,11 +50,11 @@ and binary_part =
   | Partial_signature_item of signature_item
   | Partial_module_type of module_type
 
+type relation_kind = Directed | Undirected
```

Comment:
Since this part of the `cmt_format` would be part of the external API, it is probably better to use more expressive name as suggested by @gasche . What about
```ocaml
type dependency_kind = Implementation_to_interface | Any
```
?

Reply:
Sounds reasonable, I made the change in [01d5a0d](https://github.com/ocaml/ocaml/pull/13308/commits/01d5a0d12dad43d327849ed3296e314fa524f294)

Reply:
Now that I think about it, `Structure_to_signature`  or `Definition_to_declaration` might be a better choice ? 

These does not only cover cases where the link is between the implementation file and the interface file, but also links between a submodule's structure and its signature.

Reply:
`Definition_to_declaration` sounds more future proof, and we could expand the `Any` constructor to `Declaration_to_declaration`.

Reply:
Right, I made these changes.